include/kafka/Producer.h

```diff
@@ -0,0 +1,187 @@
+#pragma once
+
+#include "kafka/Project.h"
+
+#include "kafka/ProducerRecord.h"
+#include "kafka/RdKafkaHelper.h"
+#include "kafka/Timestamp.h"
+#include "kafka/Types.h"
+
+#include "librdkafka/rdkafka.h"
+
+#include <functional>
+#include <memory>
+
+
+namespace KAFKA_API {
+
+/**
+ * Namespace for datatypes defined for KafkaProducer.
+ */
+namespace Producer
+{
+    /**
+     * The metadata for a record that has been acknowledged by the server.
+     */
+    class RecordMetadata
```

Comment:
As for consumer, it may make sense to put RecordMetadata into a separated header file

Reply:
Yeah, this block of code was moved from `KafkaProducer.h`, and eventually we'll pick a better name for `Producer.h`, e.g. `ProducerCommom.h`, or `ProducerTypes.h`, etc. 😄